README.md

```diff
@@ -696,6 +701,74 @@ api_key = "{{ env.OPENAI_API_KEY }}"
 [llm.providers.openai.models.gpt-3-5-turbo]
 ```
 
+##### Pattern-Based Model Routing
+
+Use regex patterns to dynamically match model names without explicit configuration. This is particularly useful for:
+- AI code editors (Claude Code, Cursor) that need to dynamically select models
+- Allowing users to experiment with new models without configuration changes
+- Supporting model families without listing every variant
+
+```toml
+[llm.providers.openai]
+type = "openai"
+api_key = "{{ env.OPENAI_API_KEY }}"
+model_pattern = "^gpt-.*"  # Match all models starting with "gpt-"
+
+[llm.providers.anthropic]
+type = "anthropic"
+api_key = "{{ env.ANTHROPIC_API_KEY }}"
+model_pattern = "^claude-.*"  # Match all Claude models
+
+[llm.providers.google]
+type = "google"
+api_key = "{{ env.GOOGLE_API_KEY }}"
+model_pattern = "^gemini-.*"  # Match all Gemini models
+```
+
+**Pattern Routing Features:**
+- **Case-insensitive matching**: `GPT-4` and `gpt-4` both match `^gpt-.*`
+- **Provider ordering**: When multiple patterns match, the first provider in config wins
+- **No validation required**: Nexus forwards requests directly to the provider API
+- **Model discovery**: Pattern-matched models appear in `/v1/models` responses (fetched from provider APIs)
+- **5-minute cache**: Model lists are cached and reused if provider APIs are unavailable
+
+**Pattern Rules:**
+- Cannot be empty
+- Cannot contain `/` characters
+- Must be valid regex syntax
```

Comment:
We should link to [the crate docs](https://docs.rs/regex/1.11.3/regex/) to document which regex syntax variant we are talking about.